src/replication.c

```diff
@@ -2679,21 +2689,26 @@ replicate(ISTGT_LU_DISK *spec, ISTGT_LU_CMD_Ptr cmd, uint64_t offset, uint64_t n
 				goto retry_read;
 			}
 
+
+			timesdiff(CLOCK_MONOTONIC, queued_time, now, diff);
 			count = 0;
-			if (success_resp == true) {
-				if (extra_wait.tv_sec == 0)
-					clock_gettime(CLOCK_REALTIME,
-					    &extra_wait);
-				clock_gettime(CLOCK_REALTIME, &now);
+			if (rcomm_cmd->opcode == ZVOL_OPCODE_WRITE) {
 				for (i = 0; i < rcomm_cmd->copies_sent; i++) {
 					if (rcomm_cmd->resp_list[i].status &
 					    (RECEIVED_OK|RECEIVED_ERR))
 						count++;
+					else if (diff.tv_sec >= io_max_wait_time) {
+						ASSERT(rcomm_cmd->resp_list[i].replica);
```

Comment:
also, would it be better to set the rcomm_cmd->resp_list[i].status with new flag like TIMED_OUT_ERR?

Reply:
we need to be careful that this code doesn't executed twice for a replica, as, replica might have got deleted when this is executed first time for a replica.

Reply:
ok